File: src/lib/safety.ts
import { bmi } from "./science";

export type SafetyFlag = "under18" | "pregnant" | "eatingDisorder" | "diabetesMeds" | "glp1" | "lowBmi" | "heart";

export interface SafetyResult {
  block: boolean;
  notes: string[];
}

/**
 * فحص أمان قبل إعطاء عجز سعرات. سند مدرب سلوكي، ليس بديلاً عن الطبيب.
 */
export function assessSafety(input: { age: number; weightKg: number; heightCm: number; flags: SafetyFlag[] }): SafetyResult {
  const notes: string[] = [];
  let block = false;
  const b = bmi(input.weightKg, input.heightCm);

  if (input.age < 18 || input.flags.includes("under18")) {
    block = true;
    notes.push("لأقل من ١٨ سنة: الأفضل خطة يشرف عليها طبيب أطفال أو أخصائي تغذية، مو عجز سعرات.");
  }
  if (input.flags.includes("pregnant")) {
    block = true;
    notes.push("أثناء الحمل أو الرضاعة لا ننصح بعجز سعرات. سند يقدر يساعدك بعادات صحية فقط بعد موافقة طبيبتك.");
  }
  if (input.flags.includes("eatingDisorder")) {
    block = true;
    notes.push("مع تاريخ اضطراب أكل، عدّ السعرات قد يضر. نوصي بمختص نفسي/تغذية أولاً.");
  }
  if (b < 20 || input.flags.includes("lowBmi")) {
    block = true;
    notes.push("وزنك ضمن الطبيعي أو أقل؛ التنحيف غير مناسب. ركّز على القوة واللياقة.");
  }
  if (input.flags.includes("diabetesMeds")) {
    notes.push("أدوية السكري (خاصة الإنسولين والسلفونيل يوريا) قد تحتاج تعديل مع تقليل الأكل — راجع طبيبك قبل البدء.");
  }
  if (input.flags.includes("glp1")) {
    notes.push("مع أدوية GLP-1 (مثل أوزمبيك/مونجارو): البروتين وتمارين المقاومة أهم شي لحماية عضلاتك — سند يرفع أولويتهم لك.");
  }
  if (input.flags.includes("heart")) {
    notes.push("مع أمراض القلب أو الضغط غير المنضبط: خذ موافقة طبيبك على التمارين، وابدأ بالمشي.");
  }
  return { block, notes };
}

File: android/core/src/main/kotlin/app/sanad/core/Safety.kt
package app.sanad.core

data class SafetyResult(val block: Boolean, val notes: List<String>)

/** فحص أمان قبل إعطاء عجز سعرات. سند مدرب سلوكي، ليس بديلاً عن الطبيب. */
fun assessSafety(age: Int, weightKg: Double, heightCm: Double, flags: List<SafetyFlag>): SafetyResult {
    val notes = mutableListOf<String>()
    var block = false
    if (age < 18 || SafetyFlag.UNDER18 in flags) {
        block = true
        notes += "لأقل من ١٨ سنة: الأفضل خطة يشرف عليها طبيب أو أخصائي تغذية، مو عجز سعرات."
    }
    if (SafetyFlag.PREGNANT in flags) {
        block = true
        notes += "أثناء الحمل أو الرضاعة لا ننصح بعجز سعرات. سند يساعدك بعادات صحية فقط بعد موافقة طبيبتك."
    }
    if (SafetyFlag.EATING_DISORDER in flags) {
        block = true
        notes += "مع تاريخ اضطراب أكل، عدّ السعرات قد يضر. نوصي بمختص نفسي/تغذية أولاً."
    }
    if (bmi(weightKg, heightCm) < 20 || SafetyFlag.LOW_BMI in flags) {
        block = true
        notes += "وزنك ضمن الطبيعي أو أقل؛ التنحيف غير مناسب. ركّز على القوة واللياقة."
    }
    if (SafetyFlag.DIABETES_MEDS in flags) notes += "أدوية السكري قد تحتاج تعديل مع تقليل الأكل — راجع طبيبك قبل البدء."
    if (SafetyFlag.GLP1 in flags) notes += "مع إبر GLP-1: البروتين وتمارين المقاومة أهم شي لحماية عضلاتك — سند يرفع أولويتهم لك."
    if (SafetyFlag.HEART in flags) notes += "مع أمراض القلب أو الضغط غير المنضبط: خذ موافقة طبيبك على التمارين، وابدأ بالمشي."
    return SafetyResult(block, notes)
}
